impeller/typographer/backends/skia/text_render_context_skia.cc

```diff
@@ -421,32 +430,27 @@ static std::shared_ptr<Texture> UploadGlyphTextureAtlas(
   texture_descriptor.format = format;
   texture_descriptor.size = atlas_size;
 
+  // If the alignment isn't a multiple of the pixel format, we cannot use
+  // a linear texture and instead must blit to a new texture.
   if (pixmap.rowBytes() * pixmap.height() !=
       texture_descriptor.GetByteSizeOfBaseMipLevel()) {
     return nullptr;
   }
 
-  auto texture = allocator->CreateTexture(texture_descriptor);
+  FML_DCHECK(allocator.MinimumBytesPerRow(format) <= pixmap.rowBytes());
+  auto texture = device_buffer->AsTexture(allocator, texture_descriptor,
+                                          texture_descriptor.GetBytesPerRow());
```

Comment:
Does this punish non-metal implementations right now?

I think it doesn't because before we did the copy explicitly in this code, and now it gets done implicitly if needed there, right?

Reply:
Yup!